osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -354,38 +363,47 @@ func runPackageInstallFromNewRepoTest(ctx context.Context, testCase *junitxml.Te
-	packageInstallRemovalTest := junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Package no change] [%s]", path.Base(testSetup.image)))
-	packageInstallFromNewRepoTest := junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Add a new package from new repository] [%s]", path.Base(testSetup.image)))
-
 	var logwg sync.WaitGroup
 
-	for tc, f := range map[*junitxml.TestCase]func(context.Context, *junitxml.TestCase, *packageManagementTestSetup, *log.Logger, *sync.WaitGroup, *testconfig.Project){
-		createOsConfigTest:            runCreateOsConfigTest,
-		packageInstallTest:            runPackageInstallTest,
-		packageRemovalTest:            runPackageRemovalTest,
-		packageInstallRemovalTest:     runPackageInstallRemovalTest,
-		packageInstallFromNewRepoTest: runPackageInstallFromNewRepoTest,
-	} {
-		tfname := strings.ToLower(testSetup.fname)
-		ttc := strings.ToLower(strings.Replace(getTestNameFromTestCase(tc.Name), "Test", "", -1))
-		if strings.Compare(tfname, ttc) != 0 {
-			continue
-		}
-		if tc.FilterTestCase(regex) {
-			tc.Finish(tests)
-		} else {
-			logger.Printf("Running TestCase %q", tc.Name)
-			f(ctx, tc, testSetup, logger, &logwg, testProjectConfig)
-			tc.Finish(tests)
-			logger.Printf("TestCase %q finished in %fs", tc.Name, tc.Time)
-		}
+	tc, f := getTestCaseFromTestSetUp(testSetup)
+	if tc.FilterTestCase(regex) {
+		tc.Finish(tests)
+	} else {
+		logger.Printf("Running TestCase %q", tc.Name)
+		f(ctx, tc, testSetup, logger, &logwg, testProjectConfig)
+		tc.Finish(tests)
+		logger.Printf("TestCase %q finished in %fs", tc.Name, tc.Time)
 	}
+
 	logwg.Wait()
 }
 
+// factory method to get testcase from the testsetup
+func getTestCaseFromTestSetUp(testSetup *packageManagementTestSetup) (*junitxml.TestCase, func(context.Context, *junitxml.TestCase, *packageManagementTestSetup, *log.Logger, *sync.WaitGroup, *testconfig.Project)) {
+	var tc *junitxml.TestCase
+	var f func(context.Context, *junitxml.TestCase, *packageManagementTestSetup, *log.Logger, *sync.WaitGroup, *testconfig.Project)
+
+	switch testSetup.fname {
+	case createOsConfigFunction:
+		tc = junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Create OsConfig] [%s]", path.Base(testSetup.image)))
+		f = runCreateOsConfigTest
+	case packageInstallFunction:
+		tc = junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Package installation] [%s]", path.Base(testSetup.image)))
+		f = runPackageInstallTest
+	case packageRemovalFunction:
+		tc = junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Package removal] [%s]", path.Base(testSetup.image)))
+		f = runPackageRemovalTest
+	case packageInstallRemovalFunction:
+		tc = junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Package no change] [%s]", path.Base(testSetup.image)))
+		f = runPackageInstallRemovalTest
+	case packageInstallFromNewRepoFunction:
+		tc = junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[Add a new package from new repository] [%s]", path.Base(testSetup.image)))
+		f = runPackageInstallFromNewRepoTest
+	}
```

Comment:
Can we have a default case that errors early? I'd imagine it's a bug to not specify any tests.

Reply:
yeah... that makes sense since the tests are specified in a different location. 
i will add a default case. also, in that case, i would still prefer to just log  a warning and not bail out the whole execution of the test runner.